### PR TITLE
Store originator on store credit events

### PR DESCRIPTION
**Pull request by jordan-brough**
This allows admins, gift cards, reimbursements, refunds, etc to be recorded as the "originator" of a store credit event.
- Add originator_id and originator_type fields on spree_store_credit_events table
- Allow sending an "originator" property via payment method authorize/capture/purchase/void/credit calls
- Add an action_originator attr_accessor on StoreCredit that propagates to the StoreCreditEvent on save
- Send an originator to StoreCredit when redeeming a gift card
- Send an originator to StoreCredit when granting a store credit via the Admin UI

Also:
- Rename StoreCredit#authorization_code -> #action_authorization_code to match the other "action" properties
- Refactor some method calls a bit
- Refactor some specs a bit

### CODE REVIEW

**Comment by jordan-brough** [comment]
Note: This will require updating spree also (to pass the originator). But it isn't blocked by that.

**Comment by athal7** [comment]
:+1:

**Comment by jordan-brough** [comment]
@athal7 and I are discussing some possible changes.

**Comment by jordan-brough** [comment]
@athal7 refactored as discussed

File: app/models/spree/store_credit.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Spree::StoreCredit < ActiveRecord::Base
   before_validation :associate_credit_type
   after_save :store_event
 
-  attr_accessor :action, :authorization_code, :action_amount
+  attr_accessor :action, :action_amount, :action_originator, :action_authorization_code
```
Comment (athal7) [inline app/models/spree/store_credit.rb:31]: i like the renaming here :)

**Comment by athal7** [comment]
looks good to me

File: app/models/spree/store_credit.rb (hunk 1 above)
Comment (jordan-brough) [inline app/models/spree/store_credit.rb:31]: cool :)

**Pull request merged**